fc/test_support.py

```diff
@@ -81,70 +82,37 @@ def within_any_tolerance(arr1, arr2, rel_tol=None, abs_tol=None):
         within_relative_tolerance(arr1, arr2, rel_tol))
 
 
-def check_results(proto, expected_spec, data_folder, rel_tol=0.01, abs_tol=0, messages=None):
+def check_results(proto, expected_spec, data_folder, rel_tol=0.01, abs_tol=0):
     """Check protocol results against saved values.
 
     Note that if the protocol is missing expected results, this is only an error if reference results are actually
     present on disk. If no reference results are available for an 'expected' output, this indicates that the protocol is
     expected to fail (or at least, not produce this output). Similarly, it is not an error if the protocol produces
     results but no reference results are available, although we do add a warning to messages (if supplied) in this case.
 
+    An assertion will be raised if any output is outside tolerances compared to the reference results, or if an expected
+    output (with reference results available) was not produced.
+
     :param proto: an instance of fc.Protocol that (hopefully) has results available to check
     :param expected_spec: a dictionary mapping result name to number of dimensions, so we can use the correct load*
         method
     :param data_folder: location of the reference data
     :param rel_tol: relative tolerance
     :param abs_tol: absolute tolerance
-    :param messages: if provided, should be a list to which failure reports will be appended. Otherwise any failure will
-        raise AssertionError.
-    :returns: a boolean indicating whether the results matched to within tolerances, or None if failure was expected.
     """
-    results_ok = True
     for name, ndims in expected_spec.items():
         data_file = os.path.join(data_folder, 'outputs_' + name + '.csv')
         try:
             actual = proto.output_env.look_up(name)
         except KeyError:
-            if os.path.exists(data_file):
-                results_ok = False
-                if messages is not None:
-                    messages.append("Output %s not produced but reference result exists" % name)
-            elif results_ok:
-                results_ok = None  # Indicate expected failure
+            assert not os.path.exists(data_file), f"Output {name} not produced but reference result exists"
             continue  # Can't compare in this case
```

Comment:
Why isn't this one an error too?

Reply:
The absence of a reference result indicates that we're expecting the simulation to fail, or otherwise not produce the output (maybe it's optional for older models).